src/renderer/gdi/state.cpp

```diff
@@ -456,7 +445,9 @@ GdiEngine::~GdiEngine()
                                                 const SIZE cellSize,
                                                 const size_t centeringHint) noexcept
 {
-    // If the soft font is currently selected, replace it with the default font.
+    // If we previously called SelectFont(_hdcMemoryContext, _softFont), it will
+    // still hold a reference to the _softFont object we're planning to overwrite.
+    // --> First revert back to the standard _hfont, lest we have dangling pointers.

```

Comment:
With the introduction of `FontType::Undefined` I was wondering if I should remove the if condition after this comment. I realized that this code is (potentially? probably?) critical for memory safety, so wrote a "Why?" comment here, stating why this is important to keep.

Reply:
Yes. Good. Thanks.